lib/galaxy/workflow/modules.py

```diff
@@ -598,6 +598,8 @@ def _find_collections_to_match(self, progress: "WorkflowProgress", step, all_inp
                     effective_input_collection_type,
                     dataset_collection_type_descriptions,
                 )
+                if history_query.direct_match(data):
```

Comment:
This might break mapped over conditional subworkflow steps, no ? It feels like something done deliberately.